src/main/java/org/example/application/CountHealthInsuranceContributionService.java

```diff
@@ -0,0 +1,99 @@
+import lombok.Getter;
+import org.example.domain.customer.Customer;
+import org.example.domain.customer.TaxPayments.FlatTax;
+import org.example.domain.customer.TaxPayments.GeneralTax;
+import org.example.domain.customer.TaxPayments.LumpSumTax;
+import org.example.domain.money.Currency;
+import org.example.domain.money.Money;
+import org.example.port.in.invoice.CountHealthInsuranceContributionUseCase;
+
+import java.math.BigDecimal;
+import java.time.YearMonth;
+
+@AllArgsConstructor
+@Getter
+
+public class CountHealthInsuranceContributionService implements CountHealthInsuranceContributionUseCase {
+    public static final Integer FIRST_CONTRIBUTION_RANGE = 5000;
+    public static final Integer SECOND_CONTRIBUTION_RANGE = 25000;
+    public static final BigDecimal MINIMUM_WAGE = BigDecimal.valueOf(3490);
+    private CountMonthlyMonthlyRevenueService countMonthlyRevenueService;
+    private CountMonthlyIncomeService countIncomeService;
+
+    public CountHealthInsuranceContributionService(CountMonthlyMonthlyRevenueService countMonthlyMonthlyRevenueService) {
+        this.countMonthlyRevenueService = countMonthlyMonthlyRevenueService;
+    }
+
+    @Override
+    public Money calculateHealthInsuranceContribution(Customer customer, YearMonth monthToCalculateHealthInsuranceContribution) {
+
+        if (LumpSumTax.class.equals(customer.getEntrepreneurshipForm().taxPaymentForm().getClass())) {
+            BigDecimal monthlyRevenue = countMonthlyRevenueService.countMonthlyRevenue(
+                            customer.getCustomerId(),
+                            monthToCalculateHealthInsuranceContribution
+                    )
+                    .amount();
+
+            //do 5k
+            if (monthlyRevenue.compareTo(BigDecimal.valueOf(FIRST_CONTRIBUTION_RANGE)) < 0) {
+                return new Money(
+                        monthlyRevenue
+                                .multiply(BigDecimal.valueOf(0.6f))
+                                .multiply(BigDecimal.valueOf(0.09f))
+                        , Currency.PLN);
+            }
+            // 5k - 25k
+            else if (
+                    monthlyRevenue.compareTo(BigDecimal.valueOf(FIRST_CONTRIBUTION_RANGE)) >= 0 &&
+                            monthlyRevenue.compareTo(BigDecimal.valueOf(SECOND_CONTRIBUTION_RANGE)) < 0
+            ) {
+                return new Money(monthlyRevenue.multiply(BigDecimal.valueOf(0.09)), Currency.PLN);
+            }
+            //powyzej 25k
+            else {
+                return new Money(
+                        monthlyRevenue
+                                .multiply(BigDecimal.valueOf(1.8))
+                                .multiply(BigDecimal.valueOf(0.09)),
+                        Currency.PLN);
+            }
+
+        }
+
+        if (FlatTax.class.equals(customer.getEntrepreneurshipForm().taxPaymentForm().getClass())) {
+            Money monthlyIncome = countIncomeService.countMonthlyIncome(
```

Comment:
to chyba tez

Reply:
CountMonthlyIncome needs to be a service because it includes the tax, contributionHealthInsurance(if his TaxPayment form isLumpSum) and later on the vat if the customer will be an enterpreneur who can include the VAT into the costs
Resolved.